illumio_connector.py

```diff
@@ -152,6 +151,286 @@ def _handle_get_traffic_analysis(self, param):
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Encountered error creating ruleset: {}".format(e),
+            )
+
+        result = self.convert_object_to_json(rule_set, action_result)
+        action_result.add_data(result)
+        return action_result.get_status()
+
+    def handle_comma_seperated_string(self, comma_str):
+        """
+        Convert comma seperated string into list.
+
+        :param comma_str: comma seperated string
+        :return : list
+        """
+        str_to_list = [x.strip() for x in comma_str.split(",")]
+        str_to_list = list(filter(None, str_to_list))
+        return str_to_list
+
+    def _handle_create_rule(self, param):
+        providers_list = self.handle_comma_seperated_string(param["providers"])
+        consumers_list = self.handle_comma_seperated_string(param["consumers"])
+        ruleset_href = param["ruleset_href"]
+        resolve_consumers_as_list = self.handle_comma_seperated_string(
+            param["resolve_consumers_as"]
+        )
+        resolve_providers_as_list = self.handle_comma_seperated_string(
+            param["resolve_providers_as"]
+        )
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            rule = illumio.Rule.build(
+                providers=providers_list,
+                consumers=consumers_list,
+                resolve_providers_as=resolve_providers_as_list,
+                resolve_consumers_as=resolve_consumers_as_list,
+                ingress_services=[],
+            )
+            rule = self._pce.rules.create(rule, parent=ruleset_href)
+        except IllumioException as e:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                "Encountered error creating rule: {}".format(e),
+            )
+
+        result = self.convert_object_to_json(rule, action_result)
+        action_result.add_data(result)
+        return action_result.set_status(
+            phantom.APP_SUCCESS, "Successfully created rule"
+        )
+
+    def _handle_create_service_binding(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        workload_hrefs_list = self.handle_comma_seperated_string(
+            param["workload_hrefs"]
+        )
+        virtual_service_href = param["virtual_service_href"]
+
+        ret_val = self.connect_pce(action_result)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        try:
+            service_binding_list_for_virtual_service = self._pce.service_bindings.get(
+                params={"virtual_service": virtual_service_href}
+            )
+            bounded_workload_with_virtual_service_list = [
+                service_binding.to_json()["workload"]["href"]
```

Comment:
Why convert to JSON here? `service_binding.workload.href` should work just as well.

Reply:
Done